onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorQAttention.cpp

```diff
@@ -0,0 +1,638 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "precomp.h"
+
+/*
+Abbreviations: B is batch_size, S is sequence_length, W is hidden_size
+               N is number of attention heads, H is head size, and W=N*H
+               M is mask_index tensor
+
+     M               A  B      C    // M, A, B, and C are Inputs
+     |               |  |     /
+     |            Dequantize /
+     |                \ |   /
+     |                 Gemm
+     |                / |   \
+     |               /  |    \
+     |              /   |     \
+     |          Slice  Slice  Slice
+     |            |     |       |
+     |            |     |       |
+     |      Identity Identity Identity // The identities are used to transpose NCHW -> NHCW while
+     |            |     |       |      // keeping the GEMM strides as NCHW to better target metacommands
+     |            |     |       |
+     ----------------- MHA -----
+                        |
+                        |
+                      Output  // Final output
+
+ This kernel creates a DML_GRAPH, as mentioned above.
+ For reference, refer to this Doc:
+ https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#com.microsoft.Attention
+ */
```

Comment:
Should it be "[https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#commicrosoftqattention](https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#commicrosoftqattention)"?